### PR TITLE
go.mod: bump baggageclaim

### LINKED ISSUE #4621: fly execute sometimes returns 500 error when loading an input directory containing a terraform state file
## Bug Report

We were trying to run `fly execute -i terraform-state-file=$TERRAFORM_PATH` and we receive 500 errors around 80% of the time because it fails to upload the input directory. The other 20% of the time, the input directory uploads successfully.

## Steps to Reproduce

1. Run Concourse on Kubernetes.

1. Deploy Ops Manager on AWS  using terraform by following the instructions here: https://docs.pivotal.io/pivotalcf/2-6/om/aws/prepare-env-terraform.html

1. After deploying Ops Manager, you should end up with a `terraform.tfstate` file.

1. Put this `terraform.tfstate` file in an empty directory.

1. Use this directory as an input into a Concourse task (for example, name it `terraform-state-file`).

1. Run `fly execute -i terraform-state-file=$PATH_TO_DIRECTORY_ABOVE`.

## Expected Results

Input directory should upload successfully and say "Done".

## Actual Results

Uploading fails 80% of the time with 500 error.

## Additional Context

Does not happen with Bosh-deployed Concourse, error only occurs with Kubernetes-deployed Concourse.

We found a workaround for this issue. By creating a copy of the input directory inside the input directory, it uploads successfully 100% of the time. More specifically:

This **doesn't work**:
```
terraform-state-file/
|-- terraform.tfstate
```

This **does work**:
```
terraform-state-file/
|-- terraform.tfstate
|-- terraform-state-file/
    |-- terraform.tfstate
```

## Version Info

* Concourse version: 5.6.0
* Deployment type (BOSH/Docker/binary): Kubernetes (Helm)
* Infrastructure/IaaS: GCP
* Browser (if applicable): Chrome
* Did this used to work? Not sure.

### LINKED ISSUE #4964: testflight's test for mounting input in the same place as output fails to materialize volume
### Bug Report

As observed by @vito and @clarafu, `testflight` running on nci.concourse-ci.org
recently started having the test where a task has the path of an `input` the
same as an `output`:

https://github.com/concourse/concourse/blob/f8ae6998684f05a1f93e0b9b3be03ab6f20007d5/testflight/volume_mounting_test.go#L39-L43

https://github.com/concourse/concourse/blob/f8ae6998684f05a1f93e0b9b3be03ab6f20007d5/testflight/fixtures/volume-mounting.yml#L175-L180

https://nci.concourse-ci.org/teams/main/pipelines/prs/jobs/testflight/builds/283#L5e0e4e39:1220

The timing of the failures (since this morning - 2020/01/03) seem to be quite
related to the moment of the bump from GKE 1.14.x to 1.15.4 which incurred in a
bump from Linux 4.15.x to 5.0.0.

Looking inside the task and inspecting the logs from the worker:

```json
{
  "timestamp": "2020-01-03T18:42:15.750634604Z",
  "level": "error",
  "source": "baggageclaim",
  "message": "baggageclaim.api.volume-server.create-volume-async.create-volume.failed-to-materialize-strategy",
  "data": {
    "error": "device or resource busy",
    "handle": "4af3ce40 -b08e-4481-6532-c63afd635f44",
    "session": "3.1.4500.1"
  }
}
```

```json
{
  "timestamp": "2020-01-03T18:42:15.750705655Z",
  "level": "error",
  "source": "baggageclaim",
  "message": "baggageclaim.api.volume-server.create-volume-async.failed-to-create",
  "data": {
    "error": "device or resource busy",
    "handle": "4af3ce40-b08e-4481-6532-c63afd635f44 ",
    "privileged": false,
    "session": "3.1.4500",
    "strategy": {
      "type": "cow",
      "volume": "a6ad87cf-6162-44cb-5c80-2fb8854b2e97"
    }
  }
}
```

### Steps to Reproduce

"reproduce":

1. run the `testflight` job in the pipeline
  - this will have the container running in our new linux 5.0-based machines on top
     of a privileged kubernetes pod
2. once it fails, hijack into it, run the docker daemon, and then run the test.

### Expected Results

test succeeds

### Actual Results

test fails

### Additional Context

- the test passes on Docker for Mac (Docker version 19.03.5, build 633a0ea) Linux
 4.19.76-linuxkit .
- the test passes on Linux 5.3.0-1009-gcp
- setting the same job in `nci` and `hh` then triggering the job (that fails to
  run on `testflight` works flawlessly

~my *guess* is that this is coming from a `mount(2)` that `runc` is trying to perform
when it tries to go from the OCI runtime spec to a container.~ err, this is from `baggageclaim`, so, no `runc` involved.

being from `mount(2)`, `EBUSY` would then interpreted as follows:

```

       EBUSY  An attempt was made to stack a new mount directly on top of an
              existing mount point that was created in this mount namespace
              with the same source and target.

```

~and yeah, indeed, we're trying to perform a stack mount there (as both input and
output would be mounted in the same directory, one would be "stacked" upon the
other).~

~what doesn't seem to make a lot of sense if the "with same source and target" -
why in `testflight` that'd be the case (assuming this is the problem)?~

### Version Info

* Concourse version: master (f8ae6998684f05a1f93e0b9b3be03ab6f20007d5)
* Deployment type (BOSH/Docker/binary): k8s on GKE (v1.15.4-gke.22)
  - `Linux gke-hush-house-ci-workers-79a0ea06-bkbz 5.0.0-1022-gke #22~18.04.3-Ubuntu SMP` Thu Oct 10 19:41:43 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
* Infrastructure/IaaS: GCP
* Browser (if applicable):
* Did this used to work? yes

### PR DESCRIPTION
**Pull request by cirocosta**
fix cow-cow and switch zstd lib

    i.  github.com/concourse/baggageclaim#34
    ii. github.com/concourse/baggageclaim#35

### Existing Issue

fixes #4964 
fixes #4621 

### Changes proposed in this pull request

* bump baggageclaim (see https://github.com/concourse/baggageclaim/releases/tag/v1.6.6)

### CODE REVIEW

**Comment by cirocosta** [comment]
(forgot to add release notes - marking as wip for now)

File: release-notes/latest.md [hunk 1]
```diff
@@ -31,3 +31,16 @@ pipeline on a team at the same time. #4092
 [OpenTelemetry]: https://opentelemetry.io/
 [Jaeger]: https://www.jaegertracing.io/
 [Stackdriver]: https://cloud.google.com/trace/
+
+#### <sub><sup><a name="5160" href="#5160">:link:</a></sup></sub> fix
+
+* Fix misuse of mount options when performing copy-on-write volumes based of
```
Comment (taylorsilva) [inline release-notes/latest.md]:
```suggestion
* Fix misuse of mount options when performing copy-on-write volumes based on
```

File: release-notes/latest.md [hunk 2]
```diff
@@ -31,3 +31,16 @@ pipeline on a team at the same time. #4092
...
+  other copy-on-write volumes 
+
+This case could be faced when making providing inputs and outputs with
```
Comment (taylorsilva) [inline release-notes/latest.md]:
```suggestion
This case could be faced when providing inputs and outputs with
```

File: release-notes/latest.md [hunk 3]
```diff
@@ -31,3 +31,16 @@ pipeline on a team at the same time. #4092
...
+overlapping paths.
+
+* Switch CGO-based Zstd library by a pure go one
+
+Certain payloads could make Concourse return internal errors due to possible
+errors in from the library we used before.
```
Comment (taylorsilva) [inline release-notes/latest.md]:
```suggestion
errors from the library we used before.
```

File: go.mod [hunk 4]
```diff
@@ -98,7 +98,7 @@ require (
 	github.com/vito/houdini v1.1.1
 	github.com/vito/twentythousandtonnesofcrudeoil v0.0.0-20180305154709-3b21ad808fcb
 	go.etcd.io/bbolt v1.3.3 // indirect
-	go.opencensus.io v0.22.2
+	go.opencensus.io v0.22.2 // indirect
```
Comment (cirocosta) [inline go.mod]: @aoldershaw noticed this change 🤔  don't know why it'd become `indirect` now 🤷‍♂  

```console
$ go mod why go.opencensus.io
github.com/concourse/concourse/tracing
go.opentelemetry.io/otel/exporter/trace/stackdriver
cloud.google.com/go/trace/apiv2
cloud.google.com/go/trace/apiv2.test
cloud.google.com/go/internal/testutil
go.opencensus.io/trace
go.opencensus.io/internal
go.opencensus.io
```

**Pull request merged**